File: solutions/javascript/word-count/1/word-count.js
function Words(){}

Words.prototype.count = function(sentence){
    var senArr = sentence.replace(/\r?\n|\r|\t/g," ").trim().toLowerCase().split(" ");
    var result = new Object;
    for(var i = 0; i < senArr.length; i++){
    	var w = senArr[i];
    	result[w] = result[w] ? result[w] +1:1;
    }
    return result;
};

module.exports = Words

File: solutions/kotlin/word-count/1/src/main/kotlin/WordCount.kt
/**
 * Counts the number of words in a phrase
 */
object WordCount{
    /**
     * Counts occurrence of each word in the sentence
     * @param sentence a sentence with words
     * @return [Map] with keys with words and values as the counts of numbers
     * */
    fun phrase(sentence: String): Map<String, Int>{
        val phrase = sentence.toLowerCase().replace(Regex("[^\\w']"), " ").trim()
        val words = phrase.split(Regex("\\s+"))
        val groupedWords = words.groupBy({ w -> w })

        return groupedWords.mapValues({ it.value.size })
    }
}